### PR TITLE
Test application facade

### PR DESCRIPTION
**Pull request by wassoares**
- Refactored tests of the ApplicationFacadeTest class; 
- Changes made to the ApplicationFacade and TestUtils classes;
- Fixed an exception message into the OrderController class;
- Added new constants to Messages class;

### CODE REVIEW

**Comment by raonismaneoto** [comment]
Congratulations! The tests have the same structure all over the class, besides you've covered it all!

File: src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java [hunk 1]
```diff
@@ -1,2235 +1,1873 @@
...
+
+        // verify
+        Mockito.verify(this.facade, Mockito.times(TestUtils.RUN_ONCE))
+                .getAuthenticationFromRequester(Mockito.eq(userToken));
+        Mockito.verify(this.authorizationPlugin, Mockito.times(TestUtils.RUN_ONCE)).isAuthorized(Mockito.eq(systemUser),
+                Mockito.eq(expectedOperation));
+        Mockito.verify(cloudNamesRequest, Mockito.times(TestUtils.RUN_ONCE)).send();
+    }
+    
+    // test case: When calling the getCloudNames method without set a remote
```
[the remark this answers is not in the document]
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java]: Okay, I'll improve the description of this test case.

File: src/main/java/cloud/fogbow/ras/core/ApplicationFacade.java [hunk 2]
```diff
@@ -199,16 +216,22 @@ public void deletePublicIp(String publicIpOrderId, String userToken) throws Fogb
 
     public List<InstanceStatus> getAllInstancesStatus(String userToken, ResourceType resourceType)
             throws FogbowException {
-        SystemUser requester = AuthenticationUtil.authenticate(getAsPublicKey(), userToken);
-        this.authorizationPlugin.isAuthorized(requester, new RasOperation(Operation.GET_ALL, resourceType));
+        
+        SystemUser requester = getAuthenticationFromRequester(userToken);
```
[the remark this answers is not in the document]
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/ApplicationFacade.java]: I liked the suggestion I will rename this method.

File: src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java [hunk 3]
```diff
@@ -1,2235 +1,1873 @@
...
+        // exercise
+        this.facade.getResourceInstance(order.getId(), userToken, ResourceType.VOLUME);
+
+        // verify
+        Mockito.verify(this.facade, Mockito.times(TestUtils.RUN_ONCE))
+                .getAuthenticationFromRequester(Mockito.eq(userToken));
+        Mockito.verify(this.orderController, Mockito.times(TestUtils.RUN_ONCE)).getOrder(Mockito.eq(order.getId()));
+        Mockito.verify(this.facade, Mockito.times(TestUtils.RUN_ONCE)).authorizeOrder(Mockito.eq(systemUser),
+                Mockito.eq(TestUtils.DEFAULT_CLOUD_NAME), Mockito.eq(Operation.GET), Mockito.eq(ResourceType.VOLUME),
+
```
Comment (ANKerD) [inline src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java]: fix camelCase in this method.

File: src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java [hunk 4]
```diff
@@ -1,2235 +1,1873 @@
...
-		PowerMockito.mockStatic(ServiceAsymmetricKeysHolder.class);
-		ServiceAsymmetricKeysHolder sakHolder = Mockito.mock(ServiceAsymmetricKeysHolder.class);
-		PowerMockito.when(ServiceAsymmetricKeysHolder.getInstance()).thenReturn(sakHolder);
-		sakHolder.setPublicKeyFilePath(null);
-
-		// exercise
-		this.facade.getPublicKey();
-	}	
+    // test case: When calling the getPublicKey method with a null Public Key File
+    // Path, it must throw an UnauthorizedRequestException.
```
[the remark this answers is not in the document]
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java]: Unexpected... I'll fix it.

File: src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java (hunk 3 above)
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/ApplicationFacadeTest.java]: Okay, I'll check and fix it.

**Pull request merged**